src/qos/scheduler.js

```diff
@@ -18,7 +18,43 @@ class Scheduler {
         'index': {},
         'running': false,
         'completed': [],
-        'queues': {}
+        'queues': {},
+        'sleep': {}
+      }
+    }
+  }
+
+  doSleepingTasks () {
```

Comment:
Can you rename this to "wakeSleepingProcesses"?

Reply:
One comment for everything: I'll fix it!